precompiles/utils-macro/src/lib.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+#![crate_type = "proc-macro"]
+extern crate proc_macro;
+use proc_macro::TokenStream;
+use proc_macro2::Literal;
+use quote::{quote, quote_spanned};
+use sha3::{Digest, Keccak256};
+use std::convert::TryInto;
+use syn::{parse_macro_input, spanned::Spanned, Expr, ExprLit, Ident, ItemEnum, Lit};
+
+#[proc_macro_attribute]
+pub fn generate_function_selector(_: TokenStream, input: TokenStream) -> TokenStream {
```

Comment:
```suggestion
/// TODO
pub fn generate_function_selector(_: TokenStream, input: TokenStream) -> TokenStream {
```

Reply:
why ?

Reply:
Just because it is usually nice to have documentation for your functions. If there is a reason or convention that we shouldn't have docs on this one, I'm fine with it.

Reply:
You are right I can document this macro :)